src/components/views/dialogs/CreateRoomDialog.tsx

```diff
@@ -46,15 +46,25 @@ interface IProps {
 }
 
 interface IState {
+    // The selected room join rule.
     joinRule: JoinRule;
-    isPublic: boolean;
+    // Indicates whether the knock room is public visible. Applies only for rooms with knock join rule.
+    isPublicKnockRoom: boolean;
+    // Indicates whether end-to-end encryption is enabled for the room.
     isEncrypted: boolean;
+    // The room name.
     name: string;
+    // The room topic.
     topic: string;
+    // The room alias.
     alias: string;
+    // Indicates whether the details section is open.
     detailsOpen: boolean;
+    // Indicates whether federation is disabled for the room.
     noFederate: boolean;
+    // Indicates whether the room name is valid.
     nameIsValid: boolean;
+    // Indicates whether the user can change encryption settings for the room.
     canChangeEncryption: boolean;
```

Comment:
thank you for adding all these! for future reference they should really be `/** ... */` -style comments for future interpretation by `tsdoc`.